docs(notification): adds A11y notification docs (#587)
carbon-design-system/carbon-website: src/pages/components/notification/accessibility.mdx

The close button contains an ARIA label of `aria-label="close"`, and the icon has `aria-hidden="true"` so it is ignored by assistive technologies. The `Tab` key is used to move focus to the close button within the notification and the `Space` key or `Enter` key activate the button to close the notification.

## Accessibility considerations

1. Avoid using a toast notifiction for critical alerts or long messages because they are timed and will disappear automatically making it difficult for people with various disabilities to get the entire message. An alert that disappears too quickly can lead to failure of the optional [WCAG 2.0 success criterion 2.2.3 (AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-no-exceptions.html).

> 2. Provide a means to make nonessential alerts optional to enhance usability for people with visual and cognitive disabilities. which also make meeting the requirements of the optional [WCAG 2.0 success criterion 2.2.4 (AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-postponed.html) more difficult.

Reviewer: ```suggestion
2. Provide a means to make nonessential alerts optional to enhance usability for people with visual and cognitive disabilities. which also make meeting the requirements of the optional [WCAG 2.0 success criterion 2.2.4 (AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-postponed.html) more difficult.
```

This isn't a complete sentence, though I'm not sure what it's trying to say.

Author: @snidersd do you have any thoughts on how we can change this sentence?

Reviewer 2: @aledavila Here is a rewrite:
Provide a means to turn off nonessential alerts to enhance usability for people with visual and cognitive disabilities. Additional information is available in the [WCAG 2.0 success criterion 2.2.4 (AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-postponed.html)success criterion. Note: Providing this functionality is optional.

> 2. Provide a means to make nonessential alerts optional to enhance usability for people with visual and cognitive disabilities. which also make meeting the requirements of the optional [WCAG 2.0 success criterion 2.2.4 (AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-postponed.html) more difficult.

Reviewer: ```suggestion
2. Provide a means to turn off nonessential alerts to enhance usability for people with visual and cognitive disabilities. Additional information is available in the WCAG 2.0 success criterion 2.2.4 (AAA)success criterion. Note: Providing this functionality is optional.
```

Reviewer: @aledavila just this, ready to rock otherwise

## Accessibility testing

Accessibility issues are tracked in the [Carbon Component GitHub repository](https://github.com/carbon-design-system/carbon/milestones).

### Automated test

<Row>
  <Column noGutterSm>
    <StructuredListWrapper>
      <StructuredListHead>
        <StructuredListRow head>
          <StructuredListCell head>
            Automated test environment
          </StructuredListCell>
          <StructuredListCell head>Results</StructuredListCell>
        </StructuredListRow>
      </StructuredListHead>
      <StructuredListBody>
        <StructuredListRow>
          <StructuredListCell>
            - macOS Mojave version 10.14.6 with VoiceOver
            <br />
            - Chrome version 77.0.3865.90
            <br />
            - Dynamic Assessment Plugin version 1.8.0.0 - IBM Accessibility WCAG
            2.1 Sept. 2019 Ruleset
            <br />- Carbon React version 7.7.1
          </StructuredListCell>
          <StructuredListCell>
            <strong>DAP Test</strong>
            <br />- Violations
          </StructuredListCell>
        </StructuredListRow>
      </StructuredListBody>
    </StructuredListWrapper>
  </Column>
</Row>

> ### macOS Screen reader tests

Reviewer 3: ```suggestion
### macOS screen reader tests
```
